### DISCUSSION TITLE
How can I notify Elementor that a CSS for a post need to be regenerated?

### DISCUSSION THREAD

**Discussion by suganyaselvakumar** [elementor/elementor#30582] [Developers Community]
If a post CSS file auto-generated by Elementor in the folder wp-content/uploads/elementor/css folder is missing, it's not recreated when either a logged-in or not logged-in user displays the page

after clearing cache how to recreate css file??? is there any hooks to create?

**Comment by jhned** [comment]
It's been a while, but I found a method for this.

```php
add_action('elementor/css-file/before_enqueue', function($obj) {
	// Limit this check to 1,000 times per hour.
	$spot_check_amt = get_transient('css_spot_check_amount');
	if (!empty($spot_check_amt) && $spot_check_amt >= '1000') {
		return false;
	}

	// Prevent posts that have been checked in the last 3 hours from being checked again.
	$post_spot_check_limit = get_transient('css_spot_check_post_' . get_the_ID());
	if (!empty($post_spot_check_limit)) {
		return false;
	}

	// Elementor has different types of CSS objects; we just need to check the post.
	if (get_class($obj) !== 'Elementor\Core\Files\CSS\Post') {
		return false;
	}

	// Double-check the post ID.
	if ($obj->get_post_id() !== get_the_ID()) {
		return false;
	}

	// Get the path, check it, and regenerate.
	$path = $obj->get_path();
	if (!file_exists($path)) {
		$obj->update();
		set_transient('css_spot_check_post_' . get_the_ID(), 'regenerated', HOUR_IN_SECONDS * 3);
	} else {
		set_transient('css_spot_check_amount', $spot_check_amt + 1, HOUR_IN_SECONDS);
		set_transient('css_spot_check_post_' . get_the_ID(), 'exists', HOUR_IN_SECONDS * 3);
	}
}, 10, 1);
```

**Comment by nicholaszein** [comment]
Hello!

This is a topic more appropriate to be discussed in the [Developers Community](https://github.com/orgs/elementor/discussions/categories/developers-community?discussions_q=is%3Aopen+category%3A%22Developers+Community%22+) Discussions category.

You can also consult our Developers Documentation, in particular the [PHP Hooks](https://developers.elementor.com/docs/hooks/php/) section.

As we work on organizing our GitHub issues to better reflect our users necessities, we are migrating some issues and converting them to Discussions.

While we appreciate contributions and bug reports, we are unable to provide support for custom code.

Thank you for your understanding.

Kind regards
